File: BrownianMotion/demo2022_2D.py
from random import random
from math import fabs, sqrt

pos = [ 0, 0 ] # dos dimensiones
dur = 1000
mayor = 0
regresos = 0
print('Caminata de', dur, 'pasos')
for paso in range(dur):
    # elegir eje (vertical o horizontal)
    eje = 0 if random() < 0.5 else 1 
    if random() < 0.5:
        pos[eje] += 1
    else:
        pos[eje] -= 1
    if all([ p == 0 for p in pos ]): # si ambos son ceros
        regresos += 1 # regreso al origen
    # raiz de la suma de los cuardados de las coordenadas
    euclideana = sqrt(sum([ p**2 for p in pos ]))
    mayor = max(mayor, euclideana)
print('Fueron', regresos, 'regresos')
print('La mayor distancia era de', mayor)

File: BrownianMotion/demo2022_2D.R
pos = c(0, 0) # dos-dimensional
dur = 1000
mayor = 0
regresos = 0
cat('Caminata de', dur, 'pasos', '\n')
for (paso in 1:dur) {
    eje = 0
    if (runif(1) < 0.5) {
        eje = 1 # horizontal
    } else {
        eje = 2 # vertical
    }
    if (runif(1) < 0.5) {
        pos[eje] = pos[eje] + 1
    } else {
        pos[eje] = pos[eje] - 1
    }
    if (all(pos == 0)) { # si ambos son ceros
        regresos = regresos + 1 # regreso al origen
    }
    manhattan = sum(abs(pos)) # suma de los valores absolutos
    mayor = max(mayor, manhattan)
}
cat('Fueron', regresos, 'regresos', '\n')
cat('La mayor distancia (Manhattan) era de', mayor, '\n')
